src/json.hpp

```diff
@@ -2885,18 +2829,13 @@ class lexer
     /// the current character
     int current = std::char_traits<char>::eof();
 
-    /// whether get() should return the last character again
-    bool next_unget = false;
-
     /// the number of characters read
     std::size_t chars_read = 0;
-    /// the start position of the current token
-    std::size_t start_pos = 0;
+    /// raw input token string (for error messages)
+    std::vector<char> token_string = std::vector<char>();
 
     /// buffer for variable-length tokens (numbers, strings)
-    std::vector<char> yytext = std::vector<char>(1024, '\0');
-    /// current index in yytext
-    std::size_t yylen = 0;
+    std::string yytext = "";
```

Comment:
Redundant initialization here

Reply:
Yes; but we're not doing that with any of the other members; I was just trying to be consistent.

Reply:
I guess it's up to personal preference, not a big deal anyway ;)

Reply:
You are relying on the optimizer to eliminate the call to strlen() here.  They should eliminate the call and just do default initialization, but better to not make it have to do that.  I would run the benchmarks with and without this and see if there's any difference.